[WPE] WPE Platform: add WPEToplevel

carlosgcampos opened the pull request:
#### af3cd7645c2f0124130279549f392f8c89501fcb
<pre>
[WPE] WPE Platform: add WPEToplevel
<a href="https://bugs.webkit.org/show_bug.cgi?id=275805">https://bugs.webkit.org/show_bug.cgi?id=275805</a>

Reviewed by Alejandro G. Castro.

Add a new class to handle platform toplevels. This will allow us to
support multiple views in the same toplevel, but for now only one is
supported.

* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::m_backend):
(WKWPE::viewToplevelIsFullScreen):
(WKWPE::View::enterFullScreen):
(WKWPE::View::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp:
(wpeDisplayQtQuickCreateView):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp: Added.
(wpeToplevelQtQuickDispose):
(wpeToplevelQtQuickResize):
(wpe_toplevel_qtquick_class_init):
(wpe_toplevel_qtquick_new):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.h: Copied from Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.cpp:
(wpe_view_qtquick_class_init):
(wpeViewQtQuickResize): Deleted.
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEToplevel.cpp: Added.
(wpeToplevelGetProperty):
(wpe_toplevel_class_init):
(wpeToplevelAddView):
(wpeToplevelRemoveView):
(wpe_toplevel_get_display):
(wpe_toplevel_set_title):
(wpe_toplevel_get_max_views):
(wpe_toplevel_get_n_views):
(wpe_toplevel_foreach_view):
(wpe_toplevel_closed):
(wpe_toplevel_get_size):
(wpe_toplevel_resize):
(wpe_toplevel_resized):
(wpe_toplevel_get_state):
(wpe_toplevel_state_changed):
(wpe_toplevel_get_scale):
(wpe_toplevel_scale_changed):
(wpe_toplevel_get_monitor):
(wpe_toplevel_monitor_changed):
(wpe_toplevel_fullscreen):
(wpe_toplevel_unfullscreen):
(wpe_toplevel_maximize):
(wpe_toplevel_unmaximize):
(wpe_toplevel_get_preferred_dma_buf_formats):
(wpe_toplevel_preferred_dma_buf_formats_changed):
* Source/WebKit/WPEPlatform/wpe/WPEToplevel.h: Added.
* Source/WebKit/WPEPlatform/wpe/WPEToplevelPrivate.h: Copied from Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h.
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpeViewGetProperty):
(wpeViewDispose):
(wpe_view_class_init):
(wpeViewToplevelStateChanged):
(wpeViewScaleChanged):
(wpeViewMonitorChanged):
(wpeViewPreferredDMABufFormatsChanged):
(wpe_view_get_toplevel):
(wpe_view_set_toplevel):
(wpe_view_get_toplevel_state):
(wpe_view_get_monitor):
(wpe_view_get_preferred_dma_buf_formats):
(wpe_view_resize): Deleted.
(wpe_view_scale_changed): Deleted.
(wpe_view_get_state): Deleted.
(wpe_view_state_changed): Deleted.
(wpe_view_fullscreen): Deleted.
(wpe_view_unfullscreen): Deleted.
(wpe_view_maximize): Deleted.
(wpe_view_unmaximize): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/WPEViewPrivate.h: Copied from Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h.
* Source/WebKit/WPEPlatform/wpe/drm/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(wpeDisplayDRMCreateView):
* Source/WebKit/WPEPlatform/wpe/drm/WPEToplevelDRM.cpp: Added.
(wpeToplevelDRMConstructed):
(wpeToplevelDRMGetMonitor):
(wpe_toplevel_drm_class_init):
(wpe_toplevel_drm_new):
* Source/WebKit/WPEPlatform/wpe/drm/WPEToplevelDRM.h: Copied from Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h.
* Source/WebKit/WPEPlatform/wpe/drm/WPEViewDRM.cpp:
(wpeViewDRMConstructed):
(wpe_view_drm_class_init):
(wpeViewDRMGetMonitor): Deleted.
* Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h:
* Source/WebKit/WPEPlatform/wpe/headless/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
(wpeDisplayHeadlessCreateView):
* Source/WebKit/WPEPlatform/wpe/headless/WPEToplevelHeadless.cpp: Added.
(wpeToplevelHeadlessResize):
(wpeToplevelHeadlessSetFullscreen):
(wpe_toplevel_headless_class_init):
(wpe_toplevel_headless_new):
* Source/WebKit/WPEPlatform/wpe/headless/WPEToplevelHeadless.h: Copied from Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h.
* Source/WebKit/WPEPlatform/wpe/headless/WPEViewHeadless.cpp:
(wpe_view_headless_class_init):
(wpeViewHeadlessResize): Deleted.
(wpeViewHeadlessSetFullscreen): Deleted.
* Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h:
* Source/WebKit/WPEPlatform/wpe/wayland/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandCreateView):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp: Added.
(DMABufFeedback::FormatTable::FormatTable):
(DMABufFeedback::FormatTable::operator bool const):
(DMABufFeedback::FormatTable::operator=):
(DMABufFeedback::FormatTable::~FormatTable):
(DMABufFeedback::DMABufFeedback):
(DMABufFeedback::drmDeviceForUsage):
(DMABufFeedback::drmDevice const):
(DMABufFeedback::Tranche::Tranche):
(DMABufFeedback::Tranche::drmDevice const):
(DMABufFeedback::format):
(wpeToplevelWaylandSaveSize):
(wpeToplevelWaylandResized):
(wpeToplevelWaylandConstructed):
(wpeToplevelWaylandDispose):
(wpeToplevelWaylandGetMonitor):
(wpeToplevelWaylandResize):
(wpeToplevelWaylandSetFullscreen):
(wpeToplevelWaylandSetMaximized):
(wpeToplevelWaylandGetPreferredDMABufFormats):
(wpe_toplevel_wayland_class_init):
(regionsEqual):
(wpeToplevelWaylandSetOpaqueRectangles):
(wpeToplevelWaylandUpdateOpaqueRegion):
(wpeToplevelWaylandFindVisibleView):
(wpeToplevelWaylandSetHasPointer):
(wpeToplevelWaylandGetVisibleViewUnderPointer):
(wpeToplevelWaylandSetIsFocused):
(wpeToplevelWaylandGetVisibleFocusedView):
(wpeToplevelWaylandViewVisibilityChanged):
(wpe_toplevel_wayland_new):
(wpe_toplevel_wayland_get_wl_surface):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.h: Copied from Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWaylandPrivate.h: Copied from Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(wpeViewWaylandConstructed):
(wpeViewWaylandDispose):
(wpeViewWaylandSetOpaqueRectangles):
(wpeViewWaylandCanBeMapped):
(wpe_view_wayland_class_init):
(wpe_view_wayland_get_wl_surface):
(): Deleted.
(DMABufFeedback::FormatTable::FormatTable): Deleted.
(DMABufFeedback::FormatTable::operator bool const): Deleted.
(DMABufFeedback::FormatTable::operator=): Deleted.
(DMABufFeedback::FormatTable::~FormatTable): Deleted.
(DMABufFeedback::DMABufFeedback): Deleted.
(DMABufFeedback::drmDeviceForUsage): Deleted.
(DMABufFeedback::drmDevice const): Deleted.
(DMABufFeedback::Tranche::Tranche): Deleted.
(DMABufFeedback::Tranche::drmDevice const): Deleted.
(DMABufFeedback::format): Deleted.
(wpeViewWaylandSaveSize): Deleted.
(wpeViewWaylandGetMonitor): Deleted.
(wpeViewWaylandResize): Deleted.
(wpeViewWaylandSetFullscreen): Deleted.
(wpeViewWaylandSetMaximized): Deleted.
(wpeViewWaylandGetPreferredDMABufFormats): Deleted.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.cpp:
(WPE::WaylandSeat::updateCursor):
(WPE::WaylandSeat::emitPointerEnter const):
(WPE::WaylandSeat::emitPointerLeave const):
(WPE::WaylandSeat::flushScrollEvent):
(WPE::WaylandSeat::handleKeyEvent):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.h:
* Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h:
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:
* Tools/MiniBrowser/wpe/main.cpp:
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/WebKitTestRunner/wpe/PlatformWebViewClientWPE.cpp:
(WTR::PlatformWebViewClientWPE::PlatformWebViewClientWPE):

Canonical link: <a href="https://commits.webkit.org/280412@main">https://commits.webkit.org/280412@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/aec8b122e3a53663ec6d094c9645b52316da9530

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56569 "2 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35896 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9042 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60180 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7010 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58696 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43519 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7201 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45821 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4902 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58599 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33748 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48820 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26682 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/56095 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30529 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6152 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6013 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/52487 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6424 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61863 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/479 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/6528 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53080 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/481 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48880 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53021 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/414 "Passed tests") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/8404 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31723 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32810 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33894 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32556 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

alexgcastro reviewed (approved):
LGTM! Great addition to the API.

carlosgcampos committed:
[WPE] WPE Platform: add WPEToplevel
https://bugs.webkit.org/show_bug.cgi?id=275805

Reviewed by Alejandro G. Castro.

Add a new class to handle platform toplevels. This will allow us to
support multiple views in the same toplevel, but for now only one is
supported.

* Source/WebKit/PlatformWPE.cmake:
* Source/WebKit/UIProcess/API/wpe/WPEWebView.cpp:
(WKWPE::m_backend):
(WKWPE::viewToplevelIsFullScreen):
(WKWPE::View::enterFullScreen):
(WKWPE::View::exitFullScreen):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEDisplayQtQuick.cpp:
(wpeDisplayQtQuickCreateView):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.cpp: Added.
(wpeToplevelQtQuickDispose):
(wpeToplevelQtQuickResize):
(wpe_toplevel_qtquick_class_init):
(wpe_toplevel_qtquick_new):
* Source/WebKit/UIProcess/API/wpe/qt6/WPEToplevelQtQuick.h: Copied from Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h.
* Source/WebKit/UIProcess/API/wpe/qt6/WPEViewQtQuick.cpp:
(wpe_view_qtquick_class_init):
(wpeViewQtQuickResize): Deleted.
* Source/WebKit/WPEPlatform/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/WPEToplevel.cpp: Added.
(wpeToplevelGetProperty):
(wpe_toplevel_class_init):
(wpeToplevelAddView):
(wpeToplevelRemoveView):
(wpe_toplevel_get_display):
(wpe_toplevel_set_title):
(wpe_toplevel_get_max_views):
(wpe_toplevel_get_n_views):
(wpe_toplevel_foreach_view):
(wpe_toplevel_closed):
(wpe_toplevel_get_size):
(wpe_toplevel_resize):
(wpe_toplevel_resized):
(wpe_toplevel_get_state):
(wpe_toplevel_state_changed):
(wpe_toplevel_get_scale):
(wpe_toplevel_scale_changed):
(wpe_toplevel_get_monitor):
(wpe_toplevel_monitor_changed):
(wpe_toplevel_fullscreen):
(wpe_toplevel_unfullscreen):
(wpe_toplevel_maximize):
(wpe_toplevel_unmaximize):
(wpe_toplevel_get_preferred_dma_buf_formats):
(wpe_toplevel_preferred_dma_buf_formats_changed):
* Source/WebKit/WPEPlatform/wpe/WPEToplevel.h: Added.
* Source/WebKit/WPEPlatform/wpe/WPEToplevelPrivate.h: Copied from Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h.
* Source/WebKit/WPEPlatform/wpe/WPEView.cpp:
(wpeViewGetProperty):
(wpeViewDispose):
(wpe_view_class_init):
(wpeViewToplevelStateChanged):
(wpeViewScaleChanged):
(wpeViewMonitorChanged):
(wpeViewPreferredDMABufFormatsChanged):
(wpe_view_get_toplevel):
(wpe_view_set_toplevel):
(wpe_view_get_toplevel_state):
(wpe_view_get_monitor):
(wpe_view_get_preferred_dma_buf_formats):
(wpe_view_resize): Deleted.
(wpe_view_scale_changed): Deleted.
(wpe_view_get_state): Deleted.
(wpe_view_state_changed): Deleted.
(wpe_view_fullscreen): Deleted.
(wpe_view_unfullscreen): Deleted.
(wpe_view_maximize): Deleted.
(wpe_view_unmaximize): Deleted.
* Source/WebKit/WPEPlatform/wpe/WPEView.h:
* Source/WebKit/WPEPlatform/wpe/WPEViewPrivate.h: Copied from Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h.
* Source/WebKit/WPEPlatform/wpe/drm/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/drm/WPEDisplayDRM.cpp:
(wpeDisplayDRMCreateView):
* Source/WebKit/WPEPlatform/wpe/drm/WPEToplevelDRM.cpp: Added.
(wpeToplevelDRMConstructed):
(wpeToplevelDRMGetMonitor):
(wpe_toplevel_drm_class_init):
(wpe_toplevel_drm_new):
* Source/WebKit/WPEPlatform/wpe/drm/WPEToplevelDRM.h: Copied from Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h.
* Source/WebKit/WPEPlatform/wpe/drm/WPEViewDRM.cpp:
(wpeViewDRMConstructed):
(wpe_view_drm_class_init):
(wpeViewDRMGetMonitor): Deleted.
* Source/WebKit/WPEPlatform/wpe/drm/wpe-drm.h:
* Source/WebKit/WPEPlatform/wpe/headless/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/headless/WPEDisplayHeadless.cpp:
(wpeDisplayHeadlessCreateView):
* Source/WebKit/WPEPlatform/wpe/headless/WPEToplevelHeadless.cpp: Added.
(wpeToplevelHeadlessResize):
(wpeToplevelHeadlessSetFullscreen):
(wpe_toplevel_headless_class_init):
(wpe_toplevel_headless_new):
* Source/WebKit/WPEPlatform/wpe/headless/WPEToplevelHeadless.h: Copied from Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h.
* Source/WebKit/WPEPlatform/wpe/headless/WPEViewHeadless.cpp:
(wpe_view_headless_class_init):
(wpeViewHeadlessResize): Deleted.
(wpeViewHeadlessSetFullscreen): Deleted.
* Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h:
* Source/WebKit/WPEPlatform/wpe/wayland/CMakeLists.txt:
* Source/WebKit/WPEPlatform/wpe/wayland/WPEDisplayWayland.cpp:
(wpeDisplayWaylandCreateView):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.cpp: Added.
(DMABufFeedback::FormatTable::FormatTable):
(DMABufFeedback::FormatTable::operator bool const):
(DMABufFeedback::FormatTable::operator=):
(DMABufFeedback::FormatTable::~FormatTable):
(DMABufFeedback::DMABufFeedback):
(DMABufFeedback::drmDeviceForUsage):
(DMABufFeedback::drmDevice const):
(DMABufFeedback::Tranche::Tranche):
(DMABufFeedback::Tranche::drmDevice const):
(DMABufFeedback::format):
(wpeToplevelWaylandSaveSize):
(wpeToplevelWaylandResized):
(wpeToplevelWaylandConstructed):
(wpeToplevelWaylandDispose):
(wpeToplevelWaylandGetMonitor):
(wpeToplevelWaylandResize):
(wpeToplevelWaylandSetFullscreen):
(wpeToplevelWaylandSetMaximized):
(wpeToplevelWaylandGetPreferredDMABufFormats):
(wpe_toplevel_wayland_class_init):
(regionsEqual):
(wpeToplevelWaylandSetOpaqueRectangles):
(wpeToplevelWaylandUpdateOpaqueRegion):
(wpeToplevelWaylandFindVisibleView):
(wpeToplevelWaylandSetHasPointer):
(wpeToplevelWaylandGetVisibleViewUnderPointer):
(wpeToplevelWaylandSetIsFocused):
(wpeToplevelWaylandGetVisibleFocusedView):
(wpeToplevelWaylandViewVisibilityChanged):
(wpe_toplevel_wayland_new):
(wpe_toplevel_wayland_get_wl_surface):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWayland.h: Copied from Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEToplevelWaylandPrivate.h: Copied from Source/WebKit/WPEPlatform/wpe/headless/wpe-headless.h.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEViewWayland.cpp:
(wpeViewWaylandConstructed):
(wpeViewWaylandDispose):
(wpeViewWaylandSetOpaqueRectangles):
(wpeViewWaylandCanBeMapped):
(wpe_view_wayland_class_init):
(wpe_view_wayland_get_wl_surface):
(): Deleted.
(DMABufFeedback::FormatTable::FormatTable): Deleted.
(DMABufFeedback::FormatTable::operator bool const): Deleted.
(DMABufFeedback::FormatTable::operator=): Deleted.
(DMABufFeedback::FormatTable::~FormatTable): Deleted.
(DMABufFeedback::DMABufFeedback): Deleted.
(DMABufFeedback::drmDeviceForUsage): Deleted.
(DMABufFeedback::drmDevice const): Deleted.
(DMABufFeedback::Tranche::Tranche): Deleted.
(DMABufFeedback::Tranche::drmDevice const): Deleted.
(DMABufFeedback::format): Deleted.
(wpeViewWaylandSaveSize): Deleted.
(wpeViewWaylandGetMonitor): Deleted.
(wpeViewWaylandResize): Deleted.
(wpeViewWaylandSetFullscreen): Deleted.
(wpeViewWaylandSetMaximized): Deleted.
(wpeViewWaylandGetPreferredDMABufFormats): Deleted.
* Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.cpp:
(WPE::WaylandSeat::updateCursor):
(WPE::WaylandSeat::emitPointerEnter const):
(WPE::WaylandSeat::emitPointerLeave const):
(WPE::WaylandSeat::flushScrollEvent):
(WPE::WaylandSeat::handleKeyEvent):
* Source/WebKit/WPEPlatform/wpe/wayland/WPEWaylandSeat.h:
* Source/WebKit/WPEPlatform/wpe/wayland/wpe-wayland.h:
* Source/WebKit/WPEPlatform/wpe/wpe-platform.h:
* Tools/MiniBrowser/wpe/main.cpp:
* Tools/Scripts/webkitpy/style/checker.py:
* Tools/WebKitTestRunner/wpe/PlatformWebViewClientWPE.cpp:
(WTR::PlatformWebViewClientWPE::PlatformWebViewClientWPE):

Canonical link: https://commits.webkit.org/280412@main
